packages/react-tooltip/Spec.md

```diff
@@ -1,63 +1,474 @@
+  - There is only ever one tooltip visible at once.
+  - The tooltip shows 250ms after the trigger element receieves either mouse/pointer hover or keyboard focus.
+    - _Exception_: if another tooltip is currently visible, the new tooltip will immediately replace the old one without any delay.
+    - _Exception_: if `onlyIfTruncated` is true, then the tooltip will only show if the target element's content is overflowing.
+  - The tooltip hides 250ms after losing mouse/pointer hover or keyboard focus (or immediately if another tooltip is shown).
+  - The tooltip hides immediately when the user presses Esc.
 
-- _Migration from v8_
-- _Migration from v0_
+- Placement
 
-## Behaviors
+  - The tooltip is placed relative to its target element, based on the `placement` property.
+  - The placement will likely be handled by the PopperJS open source package. The discussion about using Popper is in [RFC: Integrate Popper into Fluent #17209](https://github.com/microsoft/fluentui/pull/17209)
 
-_Explain how the component will behave in use, including:_
+- Focus
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+  - Content within the tooltip is not focusable, and can't be interacted with directly by keyboard or mouse.
 
-## Accessibility
+- Screen readers
+  - The tooltip is connected to the trigger element using `aria-describedby`, which should result in the screen reader reading the tooltip when shown.
 
-Base accessibility information is included in the design document. After the spec is filled and review, outcomes from it need to be communicated to design and incorporated in the design document.
+# Accessibility
 
-- Decide whether to use **native element** or folow **ARIA** and provide reasons
-- Identify the **[ARIA](https://www.w3.org/TR/wai-aria-practices-1.2/) pattern** and, if the component is listed there, follow its specification as possible.
-- Identify accessibility **variants**, the `role` ([ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#role_definitions)) of the component, its `slots` and `aria-*` props.
-- Describe the **keyboard navigation**: Tab Oder and Arrow Key Navigation. Describe any other keyboard **shortcuts** used
-- Specify texts for **state change announcements** - [ARIA live regions
-  ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (number of available items in dropdown, error messages, confirmations, ...)
-- Identify UI parts that appear on **hover or focus** and specify keyboard and screen reader interaction with them
-- List cases when **focus** needs to be **trapped** in sections of the UI (for dialogs and popups or for hierarchical navigation)
-- List cases when **focus** needs to be **moved programatically** (if parts of the UI are appearing/disappearing or other cases)
+- ARIA design pattern: [Tooltip Widget](https://www.w3.org/TR/wai-aria-practices-1.2/#tooltip)
+- The Tooltip root element will have `role="tooltip"`
```

Comment:
nit: Could we just add this to expected DOM output ? I am not in favour of specific a11y requirements for components because they should really be a requirement of delivering the component

Reply:
Yup I missed this in the DOM output; I'll add it.

> I am not in favour of specific a11y requirements for components because they should really be a requirement of delivering the component

I'm not sure I understand what you mean by this?

Reply:
Ah yeah, I think that if we need to have accessible markup for DOM output of a component let's just spec that as expected DOM output of a component and rely on a specific `Accessibility` section as little as possible.

However I do think that this might be kind of opinionated, so as I said it's really just a nit

Reply:
I think @jurokapsiar added the Accessibility section to the spec template, so he might have an opinion on this.

I think it's worthwhile to list details like this explicitly, so devs writing specs don't forget about a11y :)